src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -793,25 +944,53 @@ WEAVE_ERROR GenericConfigurationManagerImpl<ImplClass>::_ComputeProvisioningHash
         VerifyOrExit(dataBuf != NULL, err = WEAVE_ERROR_NO_MEMORY);
 
         // Read the certificate.
-        err = Impl()->_GetDeviceCertificate(dataBuf, certLen, certLen);
+        err = Impl()->_GetManufAttestDeviceCertificate(dataBuf, certLen, certLen);
         SuccessOrExit(err);
 
         // Hash the length and value of the device certificate in base-64 form.
         HashLengthAndBase64Value(hash, dataBuf, (uint16_t)certLen);
     }
 
+    // Hash the device intermediate CA certificates
+    if (Impl()->ConfigValueExists(ImplClass::kConfigKey_ManufAttestDeviceICACerts))
```

Comment:
Please update the comment above describing the input to the hash function.

We also need to update factory provisioning documentation.